src/abi/json-abi-format.md

```diff
@@ -481,125 +359,110 @@ its JSON representation would look like:
 
 ```json
 {
-  "types": [
+  "concreteTypes": [
     {
-      "typeId": "2e38e77b22c314a449e91fafed92a43826ac6aa403ae6a8acb6cf58239fbaf5d",
-      "type": "()",
-      "components": [],
-      "typeParameters": null
+      "type": "([str[5]; 3], bool, b256)",
+      "concreteTypeId": "625531542be70834dd127e771101ac1014111718451bfae996d97abe700c66a5",
+      "metadataTypeId": 1,
+    },
+    {
+      "type": "str[5]",
+      "concreteTypeId": "84877f6e98274b9e4721db68b4c0bdb9e52b8e9572c5bd7811c07a41ced882c7",
+    },
+    {
+      "type": "struct MyStruct",
+      "concreteTypeId": "392d58c694d2d91f3025f2bccfadacf2a105936f5da881b0899185d49f264522",
+      "metadataTypeId": 4,
+    },
+    {
+      "type": "u64",
+      "concreteTypeId": "1506e6f44c1d6291cdf46395a8e573276a4fa79e8ace3fc891e092ef32d1b0a0"
+    },
+    {
+      "type": "b256",
+      "concreteTypeId": "7c5ee1cecf5f8eacd1284feb5f0bf2bdea533a51e2f0c9aabe9236d335989f3b"
+    },
+    {
+      "type": "bool",
+      "concreteTypeId": "b760f44fa5965c2474a3b471467a22c43185152129295af588b022ae50b50903"
     },
     {
-      "typeId": "625531542be70834dd127e771101ac1014111718451bfae996d97abe700c66a5",
+      "type": "()",
+      "concreteTypeId": "2e38e77b22c314a449e91fafed92a43826ac6aa403ae6a8acb6cf58239fbaf5d"
+    }
+  ],
+  "metadataTypes": [
+    {
+      "metadataTypeId": 1,
```

Comment:
should we make metadataTypeId's string based in case we want to make them hashed later?